website/util/rubeus.py

```diff
@@ -165,8 +165,20 @@ def to_hgrid(self):
 
     def _collect_components(self, node, visited):
         rv = []
+        if not node.can_view(self.auth):
+            return rv
         for child in node.nodes:
-            if child.resolve()._id not in visited and not child.is_deleted and node.can_view(self.auth):
+            if child.is_deleted:
+                continue
+            elif not child.can_view(self.auth):
+                if child.primary:
+                    readable_descendants = child.find_readable_descendants(self.auth)
+                    for desc in readable_descendants:
+                        visited.append(desc.resolve()._id)
+                        rv.append(self._serialize_node(desc, visited=visited))
+                else:
```

Comment:
Minor, but I think this `else` block is superfluous. It just triggers the next iteration of this loop, but if it gets hit there's nothing else that would occur this iteration anyway.